lib/normalize_string.rb

```diff
@@ -25,13 +26,18 @@ def normalize_string_to_utf8(s, suggested_character_encoding=nil)
 
   to_try.each do |from_encoding|
     if String.method_defined?(:encode)
-      begin
-        s.force_encoding from_encoding
-        return s.encode('UTF-8') if s.valid_encoding?
-      rescue ArgumentError, Encoding::UndefinedConversionError
-        # We get this is there are invalid bytes when
-        # interpreted as from_encoding at the point of
-        # the encode('UTF-8'); move onto the next one...
+
+      if from_encoding == 'utf-7'
+        return Net::IMAP.decode_utf7(s)
+      else
+        begin
+          s.force_encoding from_encoding
+          return s.encode('UTF-8') if s.valid_encoding?
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.